File: core/src/androidTest/java/com/tunjid/androidbootstrap/core/testclasses/TestActivity.java
package com.tunjid.androidbootstrap.core.testclasses;

import android.os.Bundle;
import androidx.annotation.IdRes;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import com.tunjid.androidbootstrap.core.R;
import com.tunjid.androidbootstrap.core.abstractclasses.BaseActivity;

/**
 * Test Activity
 * <p>
 * Created by tj.dahunsi on 4/29/17.
 */
public class TestActivity extends BaseActivity {

    @IdRes public final int ignoredLayoutId = View.generateViewId();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getContentView());
    }

    private ViewGroup getContentView() {
        FrameLayout parent = new FrameLayout(this);
        FrameLayout ignored = new FrameLayout(this);
        FrameLayout inner = new FrameLayout(this);

        ignored.setId(ignoredLayoutId);
        inner.setId(R.id.main_fragment_container);

        parent.addView(ignored);
        parent.addView(inner);

        return parent;
    }
}

File: core/src/androidTest/java/com/tunjid/androidbootstrap/core/testclasses/TestActivity.kt
package com.tunjid.androidbootstrap.core.testclasses

import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import androidx.annotation.IdRes
import androidx.appcompat.app.AppCompatActivity
import com.tunjid.androidbootstrap.core.R

/**
 * Test Activity
 *
 *
 * Created by tj.dahunsi on 4/29/17.
 */
class TestActivity : AppCompatActivity() {

    @IdRes
    val ignoredLayoutId = View.generateViewId()
    val containerId = View.generateViewId()

    private val contentView: ViewGroup
        get() {
            val parent = FrameLayout(this)
            val ignored = FrameLayout(this)
            val inner = FrameLayout(this)

            ignored.id = ignoredLayoutId
            inner.id = containerId

            parent.addView(ignored)
            parent.addView(inner)

            return parent
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(contentView)
    }
}
